README.md

```diff
@@ -1,61 +1,148 @@
+parameters.
+See: [Documentation](https://www.webcomponents.org/element/@polymer/iron-location),
+  [iron-location demo](https://www.webcomponents.org/element/@polymer/iron-location/demo/demo/index.html),
+  [iron-query-params demo](https://www.webcomponents.org/element/@polymer/iron-location/demo/demo/iron-query-params.html).
 
-## &lt;iron-location&gt;
+### &lt;iron-location&gt;
 
 The `iron-location` element manages binding to and from the current URL.
 
-iron-location is the first, and lowest level element in the Polymer team's
-routing system. This is a beta release of iron-location as we continue work
-on higher level elements, and as such iron-location may undergo breaking
-changes.
-
-#### Properties
-
-When the URL is: `/search?query=583#details` iron-location's properties will be:
-
-* path: `'/search'`
-* query: `'query=583'`
-* hash: `'details'`
-
-These bindings are bidirectional. Modifying them will in turn modify the URL.
-
-iron-location is only active while it is attached to the document.
-
-#### Links
-
-While iron-location is active in the document it will intercept clicks on links
-within your site, updating the URL pushing the updated URL out through the
-databinding system. iron-location only intercepts clicks with the intent to
-open in the same window, so middle mouse clicks and ctrl/cmd clicks work fine.
-
-You can customize this behavior with the `urlSpaceRegex`.
-
-#### Dwell Time
-
-iron-location protects against accidental history spamming by only adding
-entries to the user's history if the URL stays unchanged for `dwellTime`
-milliseconds.
-
-
-
-<!-- No docs for <iron-query-params> found. -->
+### &lt;iron-query-params&gt;
+
+The `iron-query-params` element manages serialization and parsing of query
+parameter strings.
+
+## Usage
+
+### Installation
+```
+npm install --save @polymer/iron-location
+```
+
+### In an html file
+
+##### &lt;iron-location&gt;
+```html
+<html>
+  <head>
+    <script type="module">
+      import '@polymer/iron-location/iron-location.js';
+    </script>
+  </head>
+  <body>
+    <iron-location
+        path="/social/profiles"
+        hash="profilePicture"
+        query="userId=polymer&display=dark"
+        dwell-time="1000">
+    </iron-location>
+  </body>
+</html>
+```
+
+#### &lt;iron-query-params&gt;
+```html
+<html>
+  <head>
+    <script type="module">
+      import '@polymer/polymer/lib/elements/dom-bind.js';
+      import '@polymer/iron-location/iron-location.js';
+      import '@polymer/iron-location/iron-query-params.js';
+    </script>
+  </head>
+  <body>
+    <dom-bind>
+      <template>
+        <iron-location
+            path="/social/profiles"
+            hash="profilePicture"
+            query="{{paramsString}}"
+            dwell-time="1000">
+        </iron-location>
+        <iron-query-params
+            id="queryParams"
+            params-string='{{paramString}}'
+            params-object='{"userId": "polymer", "display": "dark"}'>
```

Comment:
Should this example (and the in-element one below) be setting both `params-string` and `params-object`? It seems like one will overwrite the other.

Reply:
the reason why iron-location is such a mess of a file is that it tries to resolve all the initial cases. In this case delcarative syntax should have precedent over the url. So this works as intended